packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+```
+
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-file-size` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization of the number
+
+For this we will use `Intl.NumberFormat`, passing in the value of the `lang` design token as the locale.
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be any API for getting a file size unit string for a given locale. Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+I suggest we go with option 1, primarily because I am hesitant to add twelve new label provider strings that clients are expected to localize. I suggest initially supporting the following languages:
+
+-   English ('en')
+-   German ('de')
+-   Spanish ('es')
+-   French ('fr')
+-   Italian ('it')
+-   Hebrew ('iw' or 'he')
+-   Russian ('ru')
+-   Turkish ('tr')
+-   Japanese ('ja')
+-   Chinese - simplified ('zh_CN')
+-   Chinese - traditional ('zh_TW')
+
+### Security
+
+N/A
+
+### Performance
+
+N/A
+
+### Dependencies
+
+None
+
+### Test Plan
+
+Unit tests:
+
+-   standard component tests
+-   renders blank for invalid input values: `Inf`, `-Inf`, `NaN`, and non-number
+-   for an interesting range of values, value conversion results in correct values and units (using English `lang`)
+-   correct units used for both `unit-type`s
+-   standard cell text/group header text ellipsizing tests
+-   honors `lang` value and responds to changes to `lang` value
+-   uses English labels when unsupported locale is given
+
+### Tooling
+
+N/A
+
+### Documentation
+
+This component will be documented via a new story in Storybook.
+
+---
+
+## Open Issues
+
+-   Is there a better name for this column type? E.g. `nimble-table-column-memory-size`.
```

Comment:
I was wondering the same thing, particularly because you mentioned that it doesn't need to be used specifically for file sizes. Unfortunately, I don't have a better name to suggest at this point in time.

Reply:
Changed to `nimble-table-column-byte-text`.